### ISSUE TITLE
argocd-server and argocd-repo-server do not parse SSH URLs the same ways

### ISSUE THREAD

**Issue by jannfis** [argoproj/argo-cd#2576]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: http://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

When connecting repositories via SSH with a port specification in the URL and no `ssh://` prefix, `argocd-server` correctly parses the URL, tests the connection successfully and adds the repo. However, `argocd-repo-server` cannot connect to the repository because it parses the URL differently and interprets the port spec as part of the hostname.

**To Reproduce**

- Add repository with URL `ssh://git@reposerver:22/repo` - this works with both, `argocd-server` and `argocd-repo-server`
- Add repository with URL `git@reposerver:22/repo` - this works with `argocd-server` (i.e. connection test successful, repository added) but not with `argocd-repo-server`

**Expected behavior**

Consistent behaviour with `argocd-server` and `argocd-repo-server`

**Version**

```shell
argocd-server: v1.3.0+73590e1
  BuildDate: 2019-10-27T09:25:06Z
  GitCommit: 73590e1a3982501865577c2a2c9ec8794391b813
  GitTreeState: clean
  GoVersion: go1.12.6
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: v0.13.1
  Kustomize Version: v3.1.0
  Helm Version: v2.13.1
  Kubectl Version: v1.14.0
```

**Comment by jannfis** [comment]
This seems to be already fixed with v1.3 - can't reproduce anymore.

**Comment by jannfis** [comment]
I think this has been fixed by #3062. Also, usage of SSH URLs clarified now in the docs via #3030
